nima_core/storage/process_guard.py

```
@@ -37,6 +48,138 @@
 PID_FILE = PID_DIR / "gateway.pid"
 LOCK_FD = None  # File descriptor for lock
 
+# Plugin system for memory filters
+_plugin_manager = None
+_memory_filters: List[MemoryFilterPlugin] = []
+if HAS_PLUGIN_SUPPORT:
+    try:
+        _plugin_manager = PluginManager("nima.plugins.memory_filter")
+        _load_memory_filters()
```

Ruff (0.15.5)

[error] Undefined name `_load_memory_filters`

(F821)